### ISSUE TITLE
17.3.0-preview-20220530-08 - Can not run test in Azure Pipeline

### ISSUE THREAD

**Issue by MohammadHazara** [microsoft/vstest#3737]
## Description
our pipeline have worked without any issue for many years. but started failing since this specific version 17.3.0-preview-20220530-08

## Steps to reproduce
1) install Visual Studio Test Platform (latest)
2) Run test assemblies

## Expected behavior
start running tests

## Actual behavior
Fails to execute tests

## Diagnostic logs
Starting test execution, please wait...
##[error]Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
##[error]   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.TestEngine.GetTestRuntimeProvidersForUniqueConfigurations(String runSettings, IDictionary`2 sourceToSourceDetailMap, ITestRuntimeProvider& mostRecentlyCreatedInstance)
##[error]   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.TestEngine.GetExecutionManager(IRequestData requestData, TestRunCriteria testRunCriteria, IDictionary`2 sourceToSourceDetailMap)
##[error]   at Microsoft.VisualStudio.TestPlatform.Client.TestPlatform.CreateTestRunRequest(IRequestData requestData, TestRunCriteria testRunCriteria, TestPlatformOptions options, Dictionary`2 sourceToSourceDetailMap)
##[error]   at Microsoft.VisualStudio.TestPlatform.CommandLine.TestPlatformHelpers.TestRequestManager.RunTests(IRequestData requestData, TestRunCriteria testRunCriteria, ITestRunEventsRegistrar testRunEventsRegistrar, TestPlatformOptions options, Dictionary`2 sourceToSourceDetailMap)
##[error]   at Microsoft.VisualStudio.TestPlatform.CommandLine.TestPlatformHelpers.TestRequestManager.RunTests(TestRunRequestPayload testRunRequestPayload, ITestHostLauncher testHostLauncher, ITestRunEventsRegistrar testRunEventsRegistrar, ProtocolConfig protocolConfig)
##[error]   at Microsoft.VisualStudio.TestPlatform.CommandLine.Processors.RunTestsArgumentExecutor.RunTests(String runSettings)
##[error]   at Microsoft.VisualStudio.TestPlatform.CommandLine.Processors.RunTestsArgumentExecutor.Execute()
##[error]   at Microsoft.VisualStudio.TestPlatform.CommandLine.Executor.ExecuteArgumentProcessor(IArgumentProcessor processor, Int32& exitCode)
##[error]   at Microsoft.VisualStudio.TestPlatform.CommandLine.Executor.Execute(String[] args)
##[error]   at Microsoft.VisualStudio.TestPlatform.CommandLine.Program.Main(String[] args)

## Environment
Azure Pipeline

**Comment by nohwnd** [comment]
Sorry about that. The new version finally allows running multiple target frameworks at the same time rather than unifying to a single one, and filtering out all the other dlls. 

There is new preview from yesterday, that does not have the NullReferenceException, but  suspect there is one more problem that is not covered with our tests. Could you please test it with the new preview and report what happened? 

I am suspecting your pipeline would now fail with "No suitable runtime provider found for this run.
<list of dlls>".

I will work on fixing that today.

**Comment by nohwnd** [comment]
I found the underlying issue, we don't unify to a common known framework for "unknown" frameworks anymore, so including additional dlls in the run is more likely to stop the whole run: 

OLD: The logic that forces a common framework on all dlls determines that the common framework should be .NET Core 1.0, and from that it tries to run all dlls as .NET Core. The xunit.abstractions.dll is not .NET core, it is .NET Portable, but it still does not stop the whole run, because we unify it to .NET Core, so it is only that DLL that does not run.

<img width="807" alt="image" src="https://user-images.githubusercontent.com/5735905/173514488-80c15538-f456-455a-badc-6f19515633b2.png">


NEW (broken): The new version does not unify to common framework and so the run fails because there is no run provider for .NET Portable.

<img width="487" alt="image" src="https://user-images.githubusercontent.com/5735905/173514854-f38b5240-f598-415b-b5f0-22aacfa737d4.png">


NEW (fixed): 

(No screenshot yet, because not implemented yet.)

All the dlls that can't find a run provider will report warning rather than error.

**Comment by nohwnd** [comment]
I implemented it but in any case we end up aborting the run, either early or late. It would be very useful to see what you are running into, and if your pipeline started working with the new preview.

**Comment by MohammadHazara** [comment]
Thanks for the update. I just tried with the new version. This time the null reference exception is gone but we still have problem running the tests.

Output:


/InIsolation
/Logger:"trx"
/TestAdapterPath:"D:\_work\93\s"
Starting test execution, please wait...
##[error]No suitable test runtime provider found for this run.
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.Interface.dll
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.TestFramework.dll
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.dll
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.Interface.dll
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.TestFramework.dll
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.dll
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.Interface.dll
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.TestFramework.dll
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.dll
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.Interface.dll
.
.
.
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.TestFramework.dll
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.dll
##[error]D:\_work\93\s\*PRIVATE NAMESPACE*\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.Interface.dll
Vstest.console.exe exited with code 1.
**************** Completed test execution *********************
Test results files: 
Completed TestExecution Model...
##[warning]Vstest failed with error. Check logs for failures. There might be failed tests.
##[error]Error: The process 'D:\_work\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\2.203.0\Modules\DTAExecutionHost.exe' failed with exit code 1
##[error]Vstest failed with error. Check logs for failures. There might be failed tests.

**Comment by nohwnd** [comment]
Great, thanks for confirming. I've figured out the issue and described it above, the linked PR #3760 has example that is the same as what you shown. Your run selects any *Test*.dll which, among other, includes Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.dll, which we build for .NET Portable. Before we added the possibility to run multiple target frameworks in the same run, this dlls would be "unified" to .NET Framework, and skipped. With multi-tfm, there is no unified framework, and there is no testhost provider for .NET Portable, and the run fails. This is breaking change, and I am fixing it in the linked PR.
